048-standard-error-messages/proposal.md

```diff
@@ -0,0 +1,49 @@
+# Summary
+
+This RFC proposes that the Concourse API standardize on returning human and
+machine readable error messages.
+
+Authors: @taylorsilva @aoldershaw
+
+# Motivation
+
+As a consumer of the Concourse API (e.g. `fly`) I want to return a helpful error
+messages to my end-user. This proves to be difficult to do because all I can do
+is imprecisely reason about why an error occured based on the HTTP status code
+returned by the API.
+
+Example: The API endpoint for getting a pipeline config may return HTTP 404 in
+two situations: 1) the config is not found and 2) the pipeline is archived. It's
+currently not possible without further introspecition to distinguish between
+these two cases at the client level.
+
+
+# Proposal
+
+When the Concourse API returns an error (4xx and 5xx status codes) it should
```

Comment:
Should we differentiate between different levels of verbosity for 4xx vs 5xx codes? 4xx codes usually indicate user error and misuse of the API, in which case a helpful error message would be appreciated. 5xx codes indicate something has gone wrong on the server side and there's nothing the user can really do with the error message.